### ISSUE TITLE
Google.Cloud.Datastore: x-goog-request-params issue accessing multi database 

### ISSUE THREAD

**Issue by TheSpy** [googleapis/google-cloud-dotnet#11532]
Hello,
I am using Google.Cloud.Datastore 4.7.0-beta03 to access datastore multi database.

It used to work for some time, however, today the client started throwing errors as follows:

The routing header for the request does not match the resource in the request message projects/#PROJECT_ID#/databases/#DATABASE_ID# != projects/#PROJECT_ID#.  The routing header should be x-goog-request-params:project_id=#PROJECT_ID#&database_id=#DATABASE_ID#. Please refer to https://firebase.google.com/docs/firestore/manage-databases#access_a_named_database_with_a_client_library for more details.

I use the following code to initialize db.

```
new DatastoreDbBuilder
        {
            ProjectId = #PROJECT_ID#,
            DatabaseId = #DATABASE_ID#
        }.Build()
```

What could be an issue?

**Comment by jskeet** [comment]
I'll have a look at this on Monday. Does your database ID contain any characters other than alphanumeric ones? (Just in case this is an encoding issue.)

The stack trace shows that this is in rollback - do other operations work?

**Comment by TheSpy** [comment]
Thank you!

Yes, project id and database id contains a dash in the name. 
E.g. 
Project_id: test-dev
Database_id: area-service

Same issue with other operations, e.g. upsert
![image](https://github.com/googleapis/google-cloud-dotnet/assets/514031/a9c236d4-b45b-440a-ae35-169b3d31d4b4)

**Comment by amanda-tarafa** [comment]
Also, [Google.Cloud.Datastore 4.7.0-beta03](https://www.nuget.org/packages/Google.Cloud.Datastore.V1/4.7.0-beta03) was released 4 days ago. Has it been working until today with Google.Cloud.Datastore 4.7.0-beta03 or is it possible that it was working before with Google.Cloud.Datastore 4.7.0-beta02 and it broke after you updated the dependency?

**Comment by TheSpy** [comment]
4.7.0-beta02 has been working before for a while and it broke a couple of hours ago. 
After it broke, I tried updating to later 4.7.0-beta03 version - but update didn't help. 
Same issue with 4.7.0-beta02 and 4.7.0-beta03

**Comment by amanda-tarafa** [comment]
Ok, this looks like a service issue then and possibly not related to the library itself.

Just to confirm, it had been working with the same project ID and database ID with which it is failing now?

**Comment by TheSpy** [comment]
Yes, the database with specific project_id and database_id worked and suddenly started throwing errors and it is still failing.
 
Do you need a specific project_id and database_id where it is failing now?

**Comment by amanda-tarafa** [comment]
No, I can reproduce locally, but also our integration tests that have been green till now are failing as well, in the same manner.
This is most certainly an issue with the service itself. I'll create an internal bug for them. 

I'll mark this as `external` just because there's nothing on the libraries side to be done at the moment.

**Comment by TheSpy** [comment]
Ok, thank you!
Is it possible to follow the updates on the issue anywhere?

**Comment by amanda-tarafa** [comment]
> Ok, thank you! Is it possible to follow the updates on the issue anywhere?

No, I'm sorry, the issue I've created is internal, I'll give updates as I can.

**Comment by TheSpy** [comment]
Ok, thanks!

**Comment by TheSpy** [comment]
Okay, thanks.
I found this issue from some time ago, which looks similar.
Maybe could be of any help https://github.com/googleapis/google-cloud-ruby/issues/20926

**Comment by jskeet** [comment]
I'd be surprised if it were a client issue (and if it is, the API probably needs annotating to generate the right header) but I'll have another look. (That may not be today.)

**Comment by jskeet** [comment]
Have had another look at the client library code, and we definitely populate the x-goog-request-params header as I'd expect. At the moment I still think this is a server issue.

**Comment by TheSpy** [comment]
@jskeet thanks a lot for looking into the issue

**Comment by amanda-tarafa** [comment]
@TheSpy just FYI, we now know the reason for this issue, we are looking at possible fixes. But it should be a matter of a couple of days at most.

**Comment by amanda-tarafa** [comment]
@TheSpy this should be working again now after a service side fix. I cannot longer reproduce the issue and our tests are back to green. Can you confirm it's also working for you? Thanks!

**Comment by TheSpy** [comment]
@amanda-tarafa great news! I can confirm that the issue is resolved on my side too!

Thanks a lot for helping with the issue!

**Comment by amanda-tarafa** [comment]
Amazing! I'll close this issue then.
